service/discovery/aws/compute.go

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright 2021 Fraunhofer AISEC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *           $$\                           $$\ $$\   $$\
+ *           $$ |                          $$ |\__|  $$ |
+ *  $$$$$$$\ $$ | $$$$$$\  $$\   $$\  $$$$$$$ |$$\ $$$$$$\    $$$$$$\   $$$$$$\
+ * $$  _____|$$ |$$  __$$\ $$ |  $$ |$$  __$$ |$$ |\_$$  _|  $$  __$$\ $$  __$$\
+ * $$ /      $$ |$$ /  $$ |$$ |  $$ |$$ /  $$ |$$ |  $$ |    $$ /  $$ |$$ | \__|
+ * $$ |      $$ |$$ |  $$ |$$ |  $$ |$$ |  $$ |$$ |  $$ |$$\ $$ |  $$ |$$ |
+ * \$$$$$$\  $$ |\$$$$$   |\$$$$$   |\$$$$$$  |$$ |  \$$$   |\$$$$$   |$$ |
+ *  \_______|\__| \______/  \______/  \_______|\__|   \____/  \______/ \__|
+ *
+ * This file is part of Clouditor Community Edition.
+ */
+
+package aws
+
+import (
+	"clouditor.io/clouditor/api/discovery"
+	"clouditor.io/clouditor/voc"
+	"context"
+	"errors"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/aws/aws-sdk-go-v2/service/lambda"
+	types2 "github.com/aws/aws-sdk-go-v2/service/lambda/types"
+	"github.com/aws/smithy-go"
+)
+
+// computeDiscovery handles the AWS API requests regarding the computing services (EC2 and Lambda)
+type computeDiscovery struct {
+	virtualMachineAPI EC2API
+	functionAPI       LambdaAPI
+	isDiscovering     bool
+	awsConfig         *Client
+}
+
+// EC2API describes the EC2 api interface (for mock testing)
```

Comment:
I would remove the "for mock testing" comment from here and also the others, it is a bit confusing or maybe make the comment more detailed, i.e. that this conforms to the API used by the official AWS client but can be used to mock it or something like that.

Reply:
The only reason for this interface to exist is for mock testing. Otherwise, I think, building an interface like that has no real advantage. That's why I want to put a comment there to clarify why we do it.
Maybe: "...interface which is implemented by the official AWS client and mock clients in tests"?